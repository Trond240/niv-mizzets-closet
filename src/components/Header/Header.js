import React from 'react';
import './Header.scss';
import { Link, NavLink } from 'react-router-dom';
import Search from '../../containers/Search/Search.js';

export const Header = (props) => {
  return (
  <div className='header'>
    <Link 
      to={'/'} 
      className='header__title'
      onClick={() => props.resetWithDefaultCards()}
    >
        Niv-Mizzet's Closet
    </Link>
    <Search />
    <NavLink
      to={'/favorites'}
      className='header__favorites'>
        Favorites
    </NavLink>
  </div>
  )
}
