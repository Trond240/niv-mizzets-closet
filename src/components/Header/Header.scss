@import '../../styles/mixin.scss';
@import '../../styles/variables.scss';

.header {
  @include display-flex(row nowrap, center, space-between);
  background: linear-gradient(to right, $mountain-color 30%, $primary-background 50%, $island-color 70%);
  height: 5em;
  padding: 1em;
  width: 100%;
  &__title {
    @include text-shadow(0 0 0.1em, 0.25em 0.25em 0.1em);
    color: $accent-color;
    font-family: 'Fondamento', cursive;
    font-size: 2.5em;
    margin: 0.5em;
    text-decoration: none;
  }
  &__actions {
    @include display-flex(row nowrap, center, space-around);
    width: 40%;
  }
  &__favorites,
  &__login {
    background-color: $plains-color;
    border-radius: 0.25em;
    color: $swamp-color;
    font-size: 1.5em;
    padding: 0.5em;
    text-align: center;
    text-decoration: none;
    width: 3.5em;
    &:hover {
      background-color: $forest-color;
    }
  }
}