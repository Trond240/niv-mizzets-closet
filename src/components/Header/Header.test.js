import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header.js';

describe('Header', () => {
  let wrapper;
  let passedProps = {resetWithDefaultCards: jest.fn()}

  beforeEach(() => {
    wrapper = shallow(<Header {...passedProps}/>)
  })

  it ('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  // Having same problem as in Card test where cannot mock passed method
  it ('should call the resetWithDefaultCards function when link is clicked', () => {
    wrapper.find('.header__title').simulate('click');
    expect(passedProps.resetWithDefaultCards).toHaveBeenCalled();
  })
})