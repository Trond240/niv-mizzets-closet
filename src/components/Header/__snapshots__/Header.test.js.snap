// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match a snapshot 1`] = `
<div
  className="header"
>
  <Link
    className="header__title"
    onClick={[Function]}
    to="/"
  >
    Niv-Mizzet's Closet
  </Link>
  <div
    className="header__actions"
  >
    <NavLink
      className="header__favorites"
      to="/favorites"
    >
      Favorites
    </NavLink>
    <Connect(Search) />
    <NavLink
      className="header__login"
      to="/login"
    >
      Login
    </NavLink>
  </div>
</div>
`;
